CHANGELOG.md

```diff
@@ -1,28 +1,38 @@
+* Add new ``tcp_request_parser`` and ``tcp_message_delimiter`` config option. Valid values for ``tcp_request_parser``
+include ``default`` and ``batch``. New TCP recv batch oriented request parser is much more efficient than the default
+one and should be a preferred choice in most situations.  For backward compatibility reasons, the default parser hasn't
+been changed yet.
 
 Misc:
+* On startup and when parsing a config file, agent now emits a warning if the config file is readable by others.
+* Add the config option ``enable_worker_process_metrics_gather`` to enable 'linux_process_metrics' monitor for each
+multiprocess worker.
+* Each session, which runs in a separate process, periodically writes its stats in the log file.
+The interval between writes can be changed by using the ``default_worker_session_status_message_interval``
+* Rename some of the configuration parameters: ``use_miltiprocess_copying_workers``  to ``use_multiprocess_workers``,
+``default_workers_per_api_key`` to ``default_sessions_per_api_key``. Previous option names are preserved for the
+backward compatibility but they are marked as deprecated.
+NOTE: The appropriate [environment variable names](https://app.scalyr.com/help/scalyr-agent-env-aware) are changed too.
 * Update docker monitor so we don't log some non-fatal errors under warning log level when consuming logs using Docker API.
 * Add support for ``compression_type: none`` config option which completely disables compression for outgoing requests. Right now one of the main bottle necks in the high volume scenarios in the agent is compression operation. Disabling it can, in some scenarios, lead to large increase to the overall throughput (up to 2x). Disabling the compression will in most cases result in larger data egress traffic which may incur additional charges on your infrastructure provider so this option should never be set to ``none`` unless explicitly advised by the technical support.
 * Linux system metrics monitor has been updated to also ignore ``/var/lib/docker/*`` and ``/snap/*`` mount points by default. Capturing metrics for those mount points usually offers no additional insight to the end user. For information on how to change the ignore list via configuration option, please see [RELEASE_NOTES](https://github.com/scalyr/scalyr-agent-2/blob/master/RELEASE_NOTES.md).
-
-## 2.1.16 "Lasso" - December 23, 2020
-
-<!---
-Packaged by Tomaz Muraus <tomaz@scalyr.com> on Dec 23, 2020 14:00 -0800
---->
-
-Misc:
-* On startup and when parsing a config file, agent now emits a warning if the config file is readable by others.
+* The agent install bash script now adds the Scalyr repositories directly without installing the ``scalyr-repo`` packages.
+This also eliminates errors caused by re-acquiring the package manager's lock file during the *pre/post* *install/uninstall* scripts.
+The issue occurred in both ``apt`` and ``rpm`` package managers.
```

Comment:
Because of the limitations of our packaging scripts, you can't line break lines these two lines.  They need to be part of the same line above starting with "The agent install bash script..."

Reply:
But those line breaks only for the source text, to follow the Markdown's "recommended" line length, and they have no effect in a rendered result.

https://github.com/scalyr/scalyr-agent-2/blob/160a0fb9d89de4f703bb2a31ac986f60e0c204dd/CHANGELOG.md

Reply:
The problem is that we have a custom script that converts this markdown into entries for the RPM and Debian packages changelog.  So, yeah, it is valid markdown, but it breaks our simple script.  We should fix it ... it's definitely one of those gotchas but we never got around to it.

See this for our custom script: https://github.com/scalyr/scalyr-agent-2/blob/c1218b2d0cc4f7a3217062af3e613d89c0e844a9/build_package.py#L1514

Reply:
Oh, I got in now. Thanks.